### PR TITLE
Removing references to v1beta1 from hack/ and test/

### PR DESCRIPTION
**Pull request by nikhiljindal**
For https://github.com/GoogleCloudPlatform/kubernetes/issues/6584

Running e2e tests.

### CODE REVIEW

**Comment by nikhiljindal** [comment]
e2e passed!!

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -396,22 +396,29 @@ for version in "${kube_api_versions[@]}"; do
...
+    "ports": [
       {
-          "kind": "Service",
-          "apiVersion": "v1beta1",
-          "id": "service-${version}-test",
-          "port": 80,
-          "protocol": "TCP"
+        "protocol": "TCP",
+        "port": 80,
+        "targetPort": 0
```
Comment (bgrant0607) [inline hack/test-cmd.sh]: targetPort should match the port on the targeted pods. It could be a name or number.

Comment (bgrant0607) [inline hack/test-cmd.sh]: I can't tell what this test is supposed to be doing -- it doesn't look like there are pods backing the service, at least at this point. Does targetPort = 80 work? I believe we should be defaulting targetPort to port now.

Comment (nikhiljindal) [inline hack/test-cmd.sh]: Updated the value to be 80 as discussed.
Note that there is a service definition in hack/e2e-suite/services.sh as well, which uses an explicit value of targetPort (which was taken from v1beta1.containerPort).
Does that look fine?

Comment (bgrant0607) [inline hack/test-cmd.sh]: Yes, the targetPort matches the corresponding pod'scontainerPort in that example.

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**